jerry-core/ecma/base/ecma-globals.h

```diff
@@ -55,8 +55,8 @@
  */
 typedef enum
 {
-  ECMA_TYPE_SIMPLE, /**< simple value */
-  ECMA_TYPE_NUMBER, /**< 64-bit integer */
+  ECMA_TYPE_NON_PTR_VALUE, /**< directly encoded value, a 28 bit signed integer or a simple value */
+  ECMA_TYPE_FLOAT_NUMBER, /**< pointer to a 64 or 32 bit floating point number */
```

Comment:
Terminology: to me, simpler is better. I'd suggest using `ECMA_TYPE_DIRECT` and perhaps even `ECMA_TYPE_FLOAT`. Even the comment of "non-ptr" starts explaining the type as "DIRECTly encoded"; and we haven't used "VALUE" suffices at the other labels (OBJECT, STRING) either. And its better to name something what it _is_ than what it _isn't_.

We could argue that terminology is unimportant enough not to worry about -- but using the right terms helps understanding IMHO.